validator/sawtooth_validator/journal/publisher.py

```diff
@@ -812,14 +765,81 @@ def on_check_publish_block(self, force=False):
 
     def has_batch(self, batch_id):
         with self._lock:
-            if batch_id in self._pending_batch_ids:
+            if batch_id in self._pending_batches:
                 return True
             if batch_id in self._queued_batch_ids:
                 return True
 
         return False
 
 
+class PendingBatchesPool:
+
+    def __init__(self):
+        # Ordered batches waiting to be processed
+        self._batches = list()
+        self._ids = set()
+        self._gauge = COLLECTOR.gauge('pending_batch_gauge', instance=self)
```

Comment:
In the current pattern, all the metrics are defined in the same place. Would it be better to leave the gauge in the publisher and pass it in as an argument?